src/sagemaker/jumpstart/utils.py

```diff
@@ -136,3 +144,101 @@ def is_jumpstart_model_input(model_id: Optional[str], version: Optional[str]) ->
+    region: str,
+    tolerate_vulnerable_model: Optional[bool] = None,
+    tolerate_deprecated_model: Optional[bool] = None,
+) -> JumpStartModelSpecs:
+    """Verifies that an acceptable model_id, version, scope, and region combination is provided.
+
+    Args:
+        model_id (Optional[str]): model id of the JumpStart model to verify and
+            obtains specs.
+        version (Optional[str]): version of the JumpStart model to verify and
+            obtains specs.
+        scope (Optional[str]): scope of the JumpStart model to verify.
+        region (Optional[str]): region of the JumpStart model to verify and
+            obtains specs.
+        tolerate_vulnerable_model (Optional[bool]): True if vulnerable versions of model
+            specifications should be tolerated (exception not raised). False or None, raises an
+            exception if the script used by this version of the model has dependencies with known
+            security vulnerabilities. (Default: None).
+        tolerate_deprecated_model (Optional[bool]): True if deprecated models should be tolerated
+            (exception not raised). False if these models should raise an exception.
+            (Default: None).
+
+
+    Raises:
+        NotImplementedError: If the scope is not supported.
+        ValueError: If the combination of arguments specified is not supported.
+        VulnerableJumpStartModelError: If any of the dependencies required by the script have
+            known security vulnerabilities.
+        DeprecatedJumpStartModelError: If the version of the model is deprecated.
+    """
+
+    if tolerate_vulnerable_model is None:
+        tolerate_vulnerable_model = False
+
+    if tolerate_deprecated_model is None:
+        tolerate_deprecated_model = False
+
+    if scope is None:
+        raise ValueError(
+            "Must specify `model_scope` argument to retrieve model "
+            "artifact uri for JumpStart models."
+        )
+
+    if scope not in constants.SUPPORTED_JUMPSTART_SCOPES:
+        raise NotImplementedError(
+            "JumpStart models only support scopes: "
+            f"{', '.join(constants.SUPPORTED_JUMPSTART_SCOPES)}."
+        )
+
+    assert model_id is not None
+    assert version is not None
+
+    model_specs = accessors.JumpStartModelsAccessor.get_model_specs(
+        region=region, model_id=model_id, version=version
+    )
+
+    if (
+        scope == constants.JumpStartScriptScope.TRAINING.value
+        and not model_specs.training_supported
+    ):
+        raise ValueError(
+            f"JumpStart model ID '{model_id}' and version '{version}' " "does not support training."
+        )
+
+    if model_specs.deprecated and not tolerate_deprecated_model:
+        raise DeprecatedJumpStartModelError(model_id=model_id, version=version)
+
+    if (
+        scope == constants.JumpStartScriptScope.INFERENCE.value
+        and model_specs.inference_vulnerable
+        and not tolerate_vulnerable_model
```

Comment:
if not raise `VulnerableJumpStartModelError` on `tolerate_vulnerable_model = True`, should we atleast log as a warning?